NitroxModel/Discovery/InstallationFinders/EpicGamesInstallationFinder.cs

```diff
@@ -1,37 +1,51 @@
-﻿using System;
+using NitroxModel.Discovery.Models;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Text.RegularExpressions;
 
-namespace NitroxModel.Discovery.InstallationFinders
+namespace NitroxModel.Discovery.InstallationFinders;
+
+public class EpicGamesInstallationFinder : IGameFinder
 {
-    public class EpicGamesInstallationFinder : IFindGameInstallation
+    private static readonly Regex installLocationRegex = new("\"InstallLocation\"[^\"]*\"(.*)\"", RegexOptions.Compiled);
+
+    public GameInstallation? FindGame(GameInfo gameInfo, IList<string> errors)
     {
-        private readonly Regex installLocationRegex = new("\"InstallLocation\"[^\"]*\"(.*)\"");
+        string commonAppFolder = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+        string epicGamesManifestsDir = Path.Combine(commonAppFolder, "Epic", "EpicGamesLauncher", "Data", "Manifests");
+        if (!Directory.Exists(epicGamesManifestsDir))
+        {
+            errors.Add("Epic games manifest directory does not exist. Verify that Epic Games Store has been installed.");
+            return null;
+        }
 
-        public string FindGame(IList<string> errors = null)
+        string[] files = Directory.GetFiles(epicGamesManifestsDir, "*.item");
+        foreach (string file in files)
         {
-            string epicGamesManifestsDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Epic\EpicGamesLauncher\Data\Manifests");
-            if (!Directory.Exists(epicGamesManifestsDir))
-            {
-                errors?.Add("Epic games manifest directory does not exist. Verify that Epic Games Store has been installed.");
-                return null;
-            }
+            string fileText = File.ReadAllText(file);
+            Match match = installLocationRegex.Match(fileText);
 
-            string[] files = Directory.GetFiles(epicGamesManifestsDir, "*.item");
-            foreach (string file in files)
+            if (match.Success && match.Value.Contains(gameInfo.Name))
```

Comment:
Isn't there a problem that `SubnauticaBelowZero` would also be matched with `gameInfo.Name="Subnautica"`?